app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload
+    new_status = significant_status
+    if (previous_status != new_status) # we only care if something changed
```

Comment:
You're overthinking it. :) We're sending a stream of state changes and errors. If clients have dependencies between them, they can figure things out.

In our case, I don't think it matters that you'll get 2 notifications if locutus fails.